scanner/pkg/client/client.go

```diff
@@ -37,22 +40,56 @@ type Scanner interface {
 
 // gRPCScanner A scanner client implementation based on gRPC endpoints.
 type gRPCScanner struct {
-	indexer        v4.IndexerClient
-	matcher        v4.MatcherClient
-	gRPCConnection *grpc.ClientConn
+	indexer         v4.IndexerClient
+	matcher         v4.MatcherClient
+	gRPCConnections []*grpc.ClientConn
 }
 
 // NewGRPCScanner creates a new gRPC scanner client.
 func NewGRPCScanner(ctx context.Context, opts ...Option) (Scanner, error) {
 	o := makeOptions(opts...)
+	var connList []*grpc.ClientConn
+	conn, err := createGRPCConn(ctx, o.indexerOpts)
+	if err != nil {
+		return nil, err
+	}
+	iConn, mConn := conn, conn
+	connList = append(connList, conn)
+	if !o.comboMode {
+		mConn, err = createGRPCConn(ctx, o.matcherOpts)
```

Comment:
it seems like we always try to connect to both the indexer and the matcher. What if I only want to connect to one of them? We should only attempt to connect to the ones we want to connect to

Reply:
I don't get it. We only use one connection if no specific endpoint for indexer and matcher was specified.